pyro/poutine/handlers.py

```diff
@@ -508,8 +508,11 @@ def __init__(self, rng_seed):
         super(_SeedMessenger, self).__init__()
 
     def __enter__(self):
+        self.old_state = get_rng_state()
         set_rng_seed(self.rng_seed)
-        super(_SeedMessenger, self).__enter__()
+
+    def __exit__(self, type, value, traceback):
+        set_rng_state(self.old_state)
```

Comment:
+1 on restoring the previous state - this makes the behavior identical across backends.